app/src/main/java/com/example/inharmony/ChatListAdapter.java

```diff
@@ -0,0 +1,185 @@
+            sentMessagesParseQuery.whereEqualTo("receiver", user);
+            sentMessagesParseQuery.orderByDescending("createdAt");
+
+            ParseQuery<Message> receivedMessagesParseQuery = ParseQuery.getQuery(Message.class);
+            receivedMessagesParseQuery.whereEqualTo("sender", user);
+            receivedMessagesParseQuery.whereEqualTo("receiver", ParseUser.getCurrentUser());
+            receivedMessagesParseQuery.orderByDescending("createdAt");
+
+            try {
+                sentMessages = sentMessagesParseQuery.find();
+                receivedMessages = receivedMessagesParseQuery.find();
+                Log.i("sent messages", String.valueOf(sentMessages.size()));
+                Log.i("received messages", String.valueOf(receivedMessages.size()));
+
+
+                if ((sentMessages.size() == 0) && receivedMessages.size() == 0) {
+                    tvChatPreview.setText("Start a new chat with your match!");
+                }
+                else if ((sentMessages.size() == 0) && (receivedMessages.size() > 0)) {
+                    Message message = receivedMessages.get(0);
+                    if (message.getBody() != null) {
+                        tvChatPreview.setText(user.get("name") + ": " + message.getBody());
+                    } else {
+                        tvChatPreview.setText(user.get("name") + " shared a track!");
+                    }
+                }
+                else if ((receivedMessages.size() == 0) && (sentMessages.size() > 0)) {
+                    Message message = sentMessages.get(0);
+                    if (message.getBody() != null) {
+                        tvChatPreview.setText("You: " + message.getBody());
+                    } else {
+                        tvChatPreview.setText("You shared a track.");
+                    }
+                }
+                else if ((receivedMessages.size() > 0) && (sentMessages.size() > 0)) {
+                    Date sentLatest = (Date) sentMessages.get(0).fetchIfNeeded().getCreatedAt();
+                    Log.i("date sent", sentLatest.toString());
+                    Date receivedLatest = (Date) receivedMessages.get(0).fetchIfNeeded().getCreatedAt();
+                    if (sentLatest.after(receivedLatest)) {
+                        if (sentMessages.get(0).getBody() != null) {
+                            tvChatPreview.setText("You: " + sentMessages.get(0).getBody());
+                        } else {
+                            tvChatPreview.setText("You shared a track");
+                        }
+                    } else {
+                        if (receivedMessages.get(0).getBody() != null) {
+                            tvChatPreview.setText(user.get("name") + ": " + receivedMessages.get(0).getBody());
+                        } else {
+                            tvChatPreview.setText(user.get("name") + " shared a track!");
+                        }
+                    }
```

Comment:
there are many conditions - it would be helpful to add comments that describe the scenario where each of these conditions is met (e.g., `(sentMessages.size() == 0) && receivedMessages.size() == 0)` is "Empty state")